### ISSUE TITLE
Initialize and dispose

### ISSUE THREAD

**Issue by ReindertKorver** [circus-kitchens/flutter_datawedge#24]
The readme shows:
```dart
    FlutterDataWedge dw = FlutterDataWedge(profileName: "Example Profile");
    await dw.initialize();
    StreamSubscription onScanSubscription = dw.onScanResult.listen((ScanResult result) {
        print(result.data);
    });
    
    [...]
    
    // Stop listening for new scans.
    onScanSubscription.cancel();
    dw.dispose();
```
But the initialize function isnt available for me, same for dispose. 
pubspec.yaml
```flutter_datawedge: ^1.2.0```
I have trouble with canceling and disposing the stream before opening a new page and reenabling it after.
Example:
```dart
TextButton(
  onPressed: () async {
    onScanSubscription?.cancel();
    onScanSubscription = null;
    dw = null;

    await Navigator.of(context).push(MaterialPageRoute(
      builder: (context) => SomePage(),
    ));

    dw = FlutterDataWedge(profileName: profileName);
    onScanSubscription = dw?.onScanResult.listen(onResult);
  },
  child: const Text("Open screen")),
```
Although I see the OnResult being set, using breakpoints, it isn't called after I come back from SomePage and scan using the barcode scanner.
Am I doing this the right way or is there a better way of handling this. If I keep the onScanSubscription without canceling and setting it to null, and go to a new page and come back after, nothing will happen when I scan using the barcode scanner.

**Comment by acdwisu** [comment]
- initialize function isnt available at version 1.2.0 (pub.dev), only available at version 2.0.0-dev (current repo).
- im having similar case, reuse scanner stream after navigate to another page. in my case, i use version 1.2.0 and cancel the stream subscription each time visibility lost (with the help visibility detector plugin).
- if you will you the scanner stream in another page, then imo canceling the subscription is needed.

**Comment by ReindertKorver** [comment]
I fixed my issue by making the global variables for `FlutterDataWedge` and the `StreamSubscription` instead of doing it on every page. When getting the barcode result I return like this if the currentpage is not in view:
```dart
void onResult(ScanResult result) async {
  if (!mounted || ModalRoute.of(context)?.isCurrent == false) return;
  //other code
}
```
